x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -342,10 +342,16 @@
 
     @Override
     public Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
+        if (this.enabled == false) {
+            return Collections.emptyMap();
```

Comment:
This needs some due diligence.  If there was an ingest pipeline in the cluster state containing an inference ingest processor and the cluster was restarted with `xpack.ml.enabled: false` what would happen?

The tentative plan for restoring full cluster snapshots in Cloud in the future is to disable all X-Pack plugins during the snapshot restore, which will lead to this exact situation.

If there's any doubt about what will happen it might be safer to allow the ingest processors to exist but just have them fail on every document they process (via the failure response from the infer model action) if the license is invalid.

Reply:
@droberts195 aren't `xpack.ml.enabled` and the license checks to different things? Additionally, this is exactly what the enrich project has done (if enrich is disabled, do not provide the processors). I will reach out to core features to see what they think about this.

Reply:
@droberts195 pipelines are stored as maps in the cluster state and are not fully marshaled until the pipeline is created on the ingest node. So, clusterstate restoration and pipeline instantiation are two different steps. 

> If there was an ingest pipeline in the cluster state containing an inference ingest processor and the cluster was restarted with `xpack.ml.enabled: false` what would happen?

The cluster will start up fine, the pipeline will just fail to be instantiated on the node.

Reply:
OK great.  Sounds like it's not a problem then and the code can stay as it is.